commit 638a98fef16fa15946522d4112660b4b20c0d630
Author: Richard Hughes <richard@hughsie.com>
Date:   Thu Feb 23 10:43:37 2012 +0000

    trivial: Fix compile with older versions of GLib
    
    g_mapped_file_new_from_fd() was introduced in 2.31.0 not 2.30.0 as the
    docs suggested.
    
    Resoves https://bugs.freedesktop.org/show_bug.cgi?id=46481

diff --git a/src/cd-profile.c b/src/cd-profile.c
index a292616..323588b 100644
--- a/src/cd-profile.c
+++ b/src/cd-profile.c
@@ -1066,7 +1066,17 @@ cd_profile_set_fd (CdProfile *profile,
 	}
 
 	/* create a mapped file */
+#if GLIB_CHECK_VERSION(2,31,0)
 	priv->mapped_file = g_mapped_file_new_from_fd (fd, FALSE, error);
+	if (priv->mapped_file == NULL) {
+		g_set_error (error,
+			     CD_MAIN_ERROR,
+			     CD_MAIN_ERROR_FAILED,
+			     "failed to create mapped file from fd %i",
+			     fd);
+		goto out;
+	}
+#endif
 
 	/* parse the ICC file */
 	lcms_profile = cmsOpenProfileFromStream (stream, "r");
